generator/src/main/java/com/scottlogic/deg/generator/constraints/atomic/NotConstraint.java

```diff
@@ -11,7 +11,7 @@
 public class NotConstraint implements AtomicConstraint, VisitableProfileElement {
     public final AtomicConstraint negatedConstraint;
 
-    protected NotConstraint(AtomicConstraint negatedConstraint) {
```

Comment:
Can you keep this as protected to avoid nested not's

Reply:
thanks. changed. i couldn't think of why the constructor was protected but this makes sense now. cheers.